content/blog/aws-rds-blue-green-deployment-updates/index.md

```diff
@@ -0,0 +1,91 @@
+{{% /notes %}}
+
+AWS RDS supports blue/green deployments to support database maintainence. In a blue/green deployment, you have one production (blue) and one staging (green) database. You can safely make changes to the green instance without affecting production and promote it to be the main instance. When you enable blue/green updates, Pulumi will temporarily set up a blue/green deployment for the duration of the update to minimize downtime.
+
+There are many [benefits](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/blue-green-deployments-overview.html#blue-green-deployments-benefits) to using blue/green deployments for updates, especially for critical databases that need to be kept available and safe from untested changes.
+
+## How it works
+
+By default, updates to a database will be applied in-place, meaning there may be service interruptions while a database is modified especially when making long-running updates like changing the engine version, parameter groups, or storage settings.
+
+`blueGreenUpdate` can be enabled for [Low-Downtime Updates](https://www.pulumi.com/registry/packages/aws/api-docs/rds/instance/#low-downtime-updates). When this setting is on and an update is performed, the program starts the update by creating a blue/green deployment to maintain availability. Then, it performs the requested update on the green instance, performs a guardrail check, promotes the green instance, and finally deletes the old instance and removes the blue/green deployment. This leaves just the updated database running with minimal downtime during the update while minimizing the downtime.
+
+Though this strategy typically increases the amount of time that updates need to run, it minimizes downtime for the application by keeping a current version of the database available during the update process.
+
+## How to use
+
+Please see the full example code here: [Blue/Green Deployment Updates on RDS](https://github.com/pulumi-demos/examples/tree/main/typescript/aws-rds-blue-green-updates). To use blue/green deployments for your database updates, start by creating a database and corresponding parameter group.
+
+```typescript
+const parameterGroup = new aws.rds.ParameterGroup(
+  `${baseName}logicalreplicationgroup`,
+  {
+    family: `postgres${dbVersion.split(".")[0]}`,
+    parameters: [
+      {
+        name: "rds.logical_replication",
+        value: "1",
+        applyMethod: "pending-reboot",
+      },
+    ],
+  },
+  { deleteBeforeReplace: false }
+);
+
+const database = new aws.rds.Instance(
+  baseName,
+  {
+    allocatedStorage: storageAllocation,
+    engine: "postgres",
+    engineVersion: dbVersion,
+    instanceClass: aws.rds.InstanceType.T3_Micro,
+    username: dbUsername,
+    password: dbPW,
+    storageEncrypted: true,
+    backupRetentionPeriod: 7,
+    blueGreenUpdate: { enabled: true },
+    skipFinalSnapshot: true,
+    parameterGroupName: parameterGroup.name,
+  },
+  { dependsOn: [parameterGroup] }
+);
```

Comment:
```suggestion
});
```

`dependsOn` is not necessary because `parameterGroup.name` is referenced as an input.